agent/src/beerocks/slave/tasks/task_pool.cpp

```diff
@@ -0,0 +1,53 @@
+/* SPDX-License-Identifier: BSD-2-Clause-Patent
+ *
+ * SPDX-FileCopyrightText: 2016-2020 the prplMesh contributors (see AUTHORS.md)
+ *
+ * This code is subject to the terms of the BSD+Patent license.
+ * See LICENSE file for more details.
+ */
+
+#include "task_pool.h"
+
+using namespace beerocks;
+
+bool TaskPool::add_task(std::shared_ptr<Task> new_task)
+{
+    if (!new_task) {
+        LOG(ERROR) << "new task pointer is nullptr";
+        return false;
+    }
```

Comment:
No, it doesn't fit our needs. See my comment above.
If we declare the task object in some scope, that will force us to save it somewhere.

Reply:
See my comment above.
You don't want to leave the ownership decision left alone. you really want to _decide_ who is the owner.
Only in rare cases, usually when working with multi-threading and you **really** don't know who will need the object last, you would correctly select `std::shared_ptr` - because then it is really shared between more than one, and all have real ownership of it.
In other cases it is best to decide who holds the object, who is responsible for its lifetime. All other entities in the system just declare their needs of the object (a reference, a pointer, a copy or even requesting an ownership by `&&` or `unique_ptr`)